### ISSUE TITLE
Issue with mlflow run.log_params(vars(trainer.model.args)) using trainer

### ISSUE THREAD

**Issue by enzoBouflet** [ultralytics/ultralytics#5352]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Train, Other

### Bug

Hello ! 
I'm trying to use MLFlow  with a ClassificationTrainer and YOLOv8.
However i get this error in the mlflow.py file :
![image](https://github.com/ultralytics/ultralytics/assets/114902150/bb08ac83-efbb-47fe-98bc-15c77f982a2b)

### Environment

Ultralytics YOLOv8.0.196 🚀 Python-3.9.6 torch-2.1.0+cu121 CUDA:0 (Tesla V100-PCIE-32GB, 32501MiB)
Setup complete ✅ (96 CPUs, 125.5 GB RAM, 367.8/438.1 GB disk)

OS                  Linux-5.4.0-150-generic-x86_64-with-glibc2.27
Environment         Linux
Python              3.9.6
Install             pip
RAM                 125.51 GB
CPU                 Intel Xeon Gold 6248R 3.00GHz
CUDA                12.1

matplotlib          ✅ 3.8.0>=3.3.0
numpy               ✅ 1.26.0>=1.22.2
opencv-python       ✅ 4.8.1.78>=4.6.0
pillow              ✅ 10.0.1>=7.1.2
pyyaml              ✅ 6.0.1>=5.3.1
requests            ✅ 2.31.0>=2.23.0
scipy               ✅ 1.11.3>=1.4.1
torch               ✅ 2.1.0>=1.8.0
torchvision         ✅ 0.16.0>=0.9.0
tqdm                ✅ 4.66.1>=4.64.0
pandas              ✅ 2.1.1>=1.1.4
seaborn             ✅ 0.13.0>=0.11.0
psutil              ✅ 5.9.5
py-cpuinfo          ✅ 9.0.0
thop                ✅ 0.1.1-2209072238>=0.1.1

### Minimal Reproducible Example

```
import os
from ultralytics.models.yolo.classify import ClassificationTrainer
os.environ['MLFLOW_TRACKING_URI'] = './'
args = dict(model='yolov8n-cls.pt', data='imagenet10', epochs=3)
trainer = ClassificationTrainer(overrides=args)
trainer.train()
```

### Additional

I have corrected the bug by deleting the vars() at line 41of the file :
ultralytics/utils/callbacks/mlflow.py

`run.log_params(trainer.model.args)`

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@enzoBouflet hello there,

Thank you for bringing this issue to our attention. In YOLOv8, you're attempting to log parameters of the model_args object, however it seems like MLFlow is having an issue with how these parameters are passed.

The `trainer.model.args` is an object, and `vars()` is a built-in Python function that returns the `__dict__` attribute of an object, which is a dictionary of its attributes. It looks like MLFlow’s `log_params()` function is expecting a dictionary, but it looks like it's not handling the object attributes correctly.

Your workaround by simply not using the `vars()` function, i.e., `run.log_params(trainer.model.args)`, must be logging the trainer model's args values directly without trying to convert them to a dictionary first. We're glad you've found a temporary solution, and we thank you for notifying us!

We'll definitely look into this further in the YOLOv8 team to make sure that the trainer object's parameters can be properly logged by MLFlow without causing such issues in the future.

As you expressed willingness to submit a PR, we really appreciate it. Please feel free to do so, and we'll review it accordingly. We greatly value your contribution to the YOLOv8 community!

Best Regards,
Glenn Jocher and the Ultralytics team.

**Comment by glenn-jocher** [comment]
@AyushExel @RizwanMunawar @fcakyon have you guys used MLFlow before? Maybe it's worth creating a test for it as I don't use it myself, so I never update it, and I see users occasionally submit bug reports.

**Comment by glenn-jocher** [comment]
@enzoBouflet freat news 😃! Your original issue may now be resolved ✅ in PR #5354 in commit https://github.com/ultralytics/ultralytics/pull/5354/commits/6b808ae234e8bd6233fd0c47e35cc6976a10af96. This should be live in `ultralytics 8.0.197` soon.

To get this update:
- **[Git](https://github.com/ultralytics/ultralytics)** – Run `git pull` from within your `ultralytics/` directory or run `git clone https://github.com/ultralytics/ultralytics` again
- **[Pip](https://pypi.org/project/ultralytics/)** – Update with `pip install -U ultralytics`
- **[Notebooks](https://github.com/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb)** – Check out the updated notebooks  <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/ultralytics)** – Run `sudo docker pull ultralytics/ultralytics:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and letting us know. Please confirm if this update fixes the issue for you, and don't hesitate to report any other **issues** you find or **feature requests** you may have. Happy training with YOLOv8 🚀!

**Comment by enzoBouflet** [comment]
Hello @glenn-jocher 👋!
I've tried your fix this mornings however it doesn't work ...

![image](https://github.com/ultralytics/ultralytics/assets/114902150/0422bf34-198a-40c7-bbfd-c9209f5fb7e5)

I've found two ways for repairing this bug :

`run.log_params(trainer.model.args)`
or
run.log_params(vars(trainer.args))

**Comment by enzoBouflet** [comment]
Hello @glenn-jocher 👋 !

I found a strange things in the update 8.0.197 the fix is not the same as the changes done in the PR MLflow updates #5357.
Is this normal because the fix done in the PR #5354 doesn't work at all.

**Comment by glenn-jocher** [comment]
Hello @enzoBouflet,

Thank you for bringing this to our attention. We sincerely apologize for any confusion caused by the differences between the updates in 8.0.197 and the changes detailed in PR #5357. It's quite possible that during the merging process, some discrepancies may have occurred.

We assure you that we understand the importance of resolving the MLflow logging issue and implementing an effective fix which will be reflected in the upcoming updates. Rest assured, our team is actively working to review the discrepancy you pointed out and will work towards ensuring consistency between our PRs and updates.

Please bear with us while we investigate this and please feel free to reach out if you observe any other discrepancies or face any issues. Your patience and continued support are greatly appreciated.

Best Regards,

Glenn Jocher and the Ultralytics team.
